apps/condo/domains/ticket/access/Ticket.js

```diff
@@ -31,49 +30,27 @@ async function canReadTickets ({ authentication: { item: user }, context }) {
         }
     }
 
-    const userDivisionsInfo = await getUserDivisionsInfo(context, user.id)
+    const userEmployees = await find('OrganizationEmployee', {
+        user: { id: user.id },
+        deletedAt: null,
+    })
+    if (!userEmployees || userEmployees.length === 0) return false
 
-    if (userDivisionsInfo) {
-        const { organizationsIdsWithEmployeeInDivision, divisionsPropertiesIds } = userDivisionsInfo
-
-        return {
-            OR: [
-                {
-                    AND: [
-                        {
-                            organization: {
-                                id_not_in: organizationsIdsWithEmployeeInDivision,
-                                OR: [
-                                    queryOrganizationEmployeeFor(user.id),
-                                    queryOrganizationEmployeeFromRelatedOrganizationFor(user.id),
-                                ],
-                            },
-                        },
-                    ],
-                },
-                {
-                    AND: [
-                        {
-                            organization: { id_in: organizationsIdsWithEmployeeInDivision },
-                            OR: [
-                                { property: { id_in: divisionsPropertiesIds } },
-                                { executor: { id: user.id } },
-                                { assignee: { id: user.id } },
-                            ],
-                        },
-                    ],
-                },
-            ],
-        }
-    }
+    const {
+        organizationTicketVisibilityAccessQuery,
+        propertiesTicketVisibilityAccessQuery,
+        propertyAndSpecializationVisibilityAccessQuery,
+    } = await getTicketVisibilityTypeAccessQueryForUserEmployees(userEmployees)
 
     return {
-        organization: {
-            OR: [
-                queryOrganizationEmployeeFor(user.id),
-                queryOrganizationEmployeeFromRelatedOrganizationFor(user.id),
-            ],
-        },
+        OR: [
+            { organization: queryOrganizationEmployeeFromRelatedOrganizationFor(user.id) },
+            organizationTicketVisibilityAccessQuery,
+            propertiesTicketVisibilityAccessQuery,
+            propertyAndSpecializationVisibilityAccessQuery,
+            { assignee: { id: user.id } },
```

Comment:
It means that the user can be not an employee. It's better to add test for that case.

Reply:
yes, updated access and wrote test